python/ray/tune/experiment/trial.py

```diff
@@ -515,16 +515,27 @@ def __init__(
 
         # Checkpoint config
         checkpoint_config = checkpoint_config or CheckpointConfig()
-        checkpoint_config.checkpoint_score_attribute = (
-            checkpoint_config.checkpoint_score_attribute or TRAINING_ITERATION
-        )
+        if not _use_storage_context():
+            # TODO(justinvyu): Why is this needed?
+            checkpoint_config.checkpoint_score_attribute = (
+                checkpoint_config.checkpoint_score_attribute or TRAINING_ITERATION
+            )
 
         self.checkpoint_config = checkpoint_config
 
-        self.checkpoint_manager = _CheckpointManager(
-            checkpoint_config=self.checkpoint_config,
-            delete_fn=_CheckpointDeleter(self._trainable_name(), self.runner),
-        )
+        if _use_storage_context():
+            from ray.train._internal.checkpoint_manager import (
+                _CheckpointManager as _NewCheckpointManager,
+            )
```

Comment:
This circular dependency is caused by:

`ray.train` -> `ray.train.trainer.TrainingIterator` -> `ray.tune` -> `train._internal.CheckpointManager` -> `ray.train.CheckpointConfig`

This will be fixed when `TrainingIterator` no longer needs to depend on `tune.TrainableUtil`. The code already has a todo for deletion.